# Procedure 3: Multi-Platform Precision Pathway

## Introduction

Procedure 3 aims to construct a multi-platform precision pathway for determining the prognostic outcome of melanoma patients. This mimics a clinical diagnostic pathway, where given the results of a diagnostic test, a confident decision may be made, or the patient may be referred to collect more data from a different platform.

For this analysis, two cohorts are used. Both have clinical data, and miRNA and mRNA omics data.

1. The Cancer Genome Atlas (T.C.G.A.): 62 patients. Both omics modalities use high-throughput sequencing.
2. Melanoma Institute of Australia (M.I.A.): 45 patients. Both omics modalities use probe-based microarrays.

The prognosis outcome is classified as "Good" if survival is greater than 4 years from the date of tumor banking and “Poor” if survival is less than 1 year from the date of tumor banking. Patients who do not match a “Good” or “Poor” prognosis are excluded from analysis.

## Setting up the environment and data objects

**1. Load the R packages into the R environment**

[Timing \~ 6.5s]{style="color: grey;"}

```{r message = FALSE}
library(ClassifyR)
```

`ClassifyR` is used to perform all the demonstrated analyses below.

**2. Import preprocessed datasets for analysis**

[Timing \~ 1s]{style="color: grey;"}

```{r}
melanomaTCGA <- readRDS("data/procedure3/TCGAmRNAandMicroRNA.rds")
melanomaMIA <- readRDS("data/procedure3/MIAmRNAandMicroRNA.rds")
colData(melanomaTCGA)
colData(melanomaMIA)
```

These commands read in a preprocessed MultiAssayExperiment containing two assays: miRNA and mRNA. Pairwise ratios of gene expression are used as predictive features. Patient age is at time of diagnosis and in years.

## Precision pathways and evaluations

**3. Creating a prognostic precision pathway**

[Timing \~ 260s]{style="color: grey;"}

```{r warning = FALSE}
set.seed(1)
costs <- setNames(c(50, 700, 400), c("clinical", "RNA", "miRNA"))
pathways <- precisionPathwaysTrain(melanomaTCGA, "Outcome", mode = "combinatorial", nCores = 8)
```

The `set.seed(1)` command ensures that any subsequent operations involving randomness yield consistent results across runs. The second command trains multi-platform precision pathways using the two assays given the prognostic outcomes in the "Outcome" column in `colData(melanomaTCGA)`.

```{r warning = FALSE}
predicted <- precisionPathwaysPredict(pathways, melanomaMIA, "Outcome")
predicted <- calcCostsAndPerformance(predicted, costs)
```

Now, the constructed pathways are applied to M.I.A. dataset. Finally, the costs and performance of each potential pathway is calculated. Here, the cost of using clinical, mRNA and miRNA data have been set to \$50, \$700 and \$400, respectively.

**4. Precision Pathway Evaluation**

[Timing \~ 1.4s]{style="color: grey;"}

```{r}
summary(predicted)
```

Given the costs provided earlier in `calcCostsAndPerformance`, this command outputs the summary of performance and cost of each potential pathway for evaluation. There are two possible pathways proposed: using clinical then mRNA data or using clinical then miRNA data. Both pathways achieve the same balanced accuracy of 0.75 but due to the lower cost of miRNA sequencing, the second pathway attains a higher score and is the preferable option for prognostic purposes.

```{r}
bubblePlot(predicted)
```

This bubble plot visualises the summary created above. In a situation with more potential pathways, it would allow greater ease in selecting the best pathway to optimise cost and performance. Here, clinical-miRNA would be the preferred pathway due to its lower cost while achieving the same balanced accuracy.

```{r}
flowchart(predicted, "clinical-miRNA")
```

Given that clinical-miRNA is the selected pathway, this function creates a flowchart to show the stepwise process for determining melanoma prognosis. Clinical data is first used to classify patients into three prognosis groups: • Good prognosis: 44% of patients (27 out of total 62). • Uncertain prognosis: 21% of patients (13 out of 62) • Poor prognosis: 35% of patients (22 out of 62). Thus, clinical data alone classifies 71% of patients (Good + Poor groups), while 21% requiring further assessment.

For the Uncertain prognosis group (13 patients), additional analysis using miRNA pair data refines the classification: • Good prognosis: 19% (12 patients) are identified as “Good.” • Uncertain prognosis: 0% (0 patients) remain “Uncertain.” • Poor prognosis: 2% (1 patient) is reclassified as “Poor.” This is why a potential pathway incorporating clinical, mRNA, and miRNA data was not considered, as the first two datasets alone provided a prognosis outcome for all individuals.

```{r}
strataPlot(predicted, "clinical-miRNA")
```

This function creates a plot which allow us to visualise where the prognosis went wrong. A large amount of blue in the clinical row under the green, suggests that this pathway made the most errors in prognosis when first using clinical data, as it classified many "good" prognosis patients with "poor" prognosis. However, clinical data performed comparatively much better on "poor" prognosis patients due to the lesser proportion of blue. For the individuals who continued to use miRNA, there was high sensitivity in identifying individuals with a good prognosis but its specificity was limited as some individuals with a poor prognosis were misclassified as having a good prognosis.